localstack/utils/aws/arns.py

```diff
@@ -232,9 +234,12 @@ def fix_arn(arn):
     """Function that attempts to "canonicalize" the given ARN. This includes converting
     resource names to ARNs, replacing incorrect regions, account IDs, etc."""
     if arn.startswith("arn:aws:lambda"):
-        parts = arn.split(":")
-        region = parts[3] if parts[3] in get_valid_regions() else get_region()
-        return lambda_function_arn(lambda_function_name(arn), region_name=region)
+        arn_data = parse_arn(arn)
```

Comment:
great to use the more robust `parse_arn` utils here instead of the custom split by `:`  👍